sonaranalyzer-dotnet/rspec/cs/S5042_c#.html

```diff
@@ -1,53 +1,87 @@
+lead to denial of service. A Zip bomb is usually a malicious archive file of a few kilobytes of compressed data but turned into gigabytes of
+uncompressed data, to achieve this extreme <a href="https://en.wikipedia.org/wiki/Data_compression_ratio">compression ratio</a>, attackers will
+compress irrelevant data, a long string of repeated bytes. </p>
 <h2>Ask Yourself Whether</h2>
+<p>Archives to expand are untrusted and:</p>
 <ul>
-  <li> there is no validation of the name of the archive entry </li>
-  <li> there is no validation of the effective path where the archive entry is going to be expanded </li>
-  <li> there is no validation of the size of the expanded archive entry </li>
-  <li> there is no validation of the ratio between the compressed and uncompressed archive entry </li>
+  <li> There is no validation of the number of entries in the archive. </li>
+  <li> There is no validation of the total size of the uncompressed data. </li>
+  <li> There is no validation of the ratio between the compressed and uncompressed archive entry. </li>
 </ul>
 <p>There is a risk if you answered yes to any of those questions.</p>
 <p> </p>
 <h2>Recommended Secure Coding Practices</h2>
-<p>Validate the full path of the extracted file against the full path of the directory where files are expanded:</p>
 <ul>
-  <li> the canonical path of the expanded file must start with the canonical path of the directory where files are extracted. </li>
-  <li> the name of the archive entry must not contain "..", i.e. reference to a parent directory. </li>
+  <li> Define and control the ratio between compressed and uncompressed data, in general the data compression ratio for most of the legit archives is
+  1 to 3. </li>
+  <li> Define and control the threshold for maximum total size of the uncompressed data. </li>
+  <li> Count the number of file entries extracted from the archive and abort the extraction if their number is greater than a predefined threshold, in
+  particular it's not recommended to recursively expand archive (an entry of an archive could be also an archive). </li>
 </ul>
-<p>Stop extracting the archive if any of its entries has been tainted with a directory traversal path.</p>
-<p>Define and control the ratio between compressed and uncompress bytes.</p>
-<p>Define and control the maximum allowed expanded file size.</p>
-<p>Count the number of file entries extracted from the archive and abort the extraction if their number is greater than a predefined threshold.</p>
 <h2>Sensitive Code Example</h2>
 <pre>
+FileStream zipToOpen = new FileStream(@"ZipBomb.zip", FileMode.Open);
+ZipArchive archive = new ZipArchive(zipToOpen, ZipArchiveMode.Read);
+foreach (ZipArchiveEntry entry in archive.Entries)
+{
+  entry.ExtractToFile("./output_onlyfortesting.txt", true); // Sensitive
+}
+</pre>
+<h2>Compliant Solution</h2>
+<pre>
+int THRESHOLD_ENTRIES = 10000;
```

Comment:
RE: https://jira.sonarsource.com/browse/RSPEC-5044
@eric-therond-sonarsource - should we stop raising the hotspot when seeing a compliant example? In other words, do the S5042 changes require an issue to be created?

Reply:
Eric is on vacation. @costin-zaharia-sonarsource , I guess we can move forward with this one, allow the change and eventually open an issue to update the rule if Eric says so when back from vacation.

Reply:
Just quickly checked the code and as far as I can tell we consider both `ExtractToFile(..)` and `Open()` unsafe. This means that we will raise issues also for the compliant example -> code changes are needed before updating the docs.

I will revert the changes related to this.